### ISSUE TITLE
`no-for-loop` false negative

### ISSUE THREAD

**Issue by golopot** [sindresorhus/eslint-plugin-unicorn#286]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/286/badge.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/286)

```js
/* eslint unicorn/no-for-loop: 2 */
for (let i = 0; i < arr.length; i += 1) {
  console.log(arr[i])
}
```

**Expected**:
It reports.
**Actual**:
It does not report.

This is a subtask of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/250 .

> There is a $60.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/286).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by futpib** [comment]
Isn't this report exactly inline with the purpose of the rule? (Do not use a `for` loop that can be replaced with a `for-of` loop) https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-for-loop.md

**Comment by futpib** [comment]
Oh, I see, the issue is that the rule does _not_ reported this when it could. Sorry, I misread it the first time.

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $60.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/55832243/issues/286) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/55832243/issues/286).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by golopot** [comment]
Autofix is very desirable, but it is hard to name the variable for `arr[i]`.
